pkg/rancher-components/src/components/Form/ToggleSwitch/ToggleSwitch.vue

```diff
@@ -27,14 +27,23 @@ export default {
     },
   },
   data() {
-    return { state: this.value === this.onValue };
+    return { state: false };
   },
 
   methods: {
     toggle(neu) {
       this.state = neu === null ? !this.state : neu;
       this.$emit('input', this.state ? this.onValue : this.offValue);
     }
+  },
+
+  watch: {
+    value: {
+      handler() {
+        this.state = this.value === this.onValue;
```

Comment:
Why do we have a duplication of the `value` as `state`? What is the benefit? We may probably omit code duplication.

Reply:
Once again, I agree. I'm not quite too sure what the benefits are for having this component manage its own state like this. On the other hand, I'm trying to manage the scope of changes to make reduce the change of inadvertently introducing regressions.

Reply:
With that said, I think this is a simple enough component with a limited use that it would be easy to regression test such a refactor. I just prefer to err on the side of caution and approach problems through extending existing implementations

Reply:
Ideally the scope of the test is to have certainty of the result, which gives us freedom to refactor it.
Sorry for the pressure, I'll avoid next time 😅

Reply:
My apologies. It's no my intent to discourage discourse, I will try to check my written tone next time. Perhaps it was too late in the day and that was causing me to be too terse 😅

Please, ask questions 😁

Reply:
All good and also thanks 😄 🍻